src/test/java/com/google/sps/DatastorePlaceGuideRepositoryTest.java

```diff
@@ -0,0 +1,273 @@
+                                                          DESCRIPTION);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.LENGTH_PROPERTY, LENGTH);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.IMG_KEY_PROPERTY, IMG_KEY);
+    return placeGuideEntity;
+  }
+
+  private void saveTestPlaceGuidesEntities(List<PlaceGuide> placeGuides) {
+    for (PlaceGuide placeGuide : placeGuides) {
+      datastore.put(getEntityFromPlaceGuide(placeGuide));
+    }
+  }
+
+  // Find out if the 2 lists of placeguides are equal.
+  private boolean compare(List<PlaceGuide> a, List<PlaceGuide> b) {
+    List<PlaceGuide> b_copy = new ArrayList<>(b);
+    if (a.size() != b_copy.size()) {
+      return false;
+    }
+    for (PlaceGuide a_pg : a) {
+      boolean hasEqual = false;
+      int index_b_copy = 0;
+      while (index_b_copy < b_copy.size()) {
+        if (a_pg.equals(b_copy.get(index_b_copy))) {
+          hasEqual = true;
+          b_copy.remove(index_b_copy);
+          break;
+        }
+        index_b_copy++;
+      }
+      if (!hasEqual) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private DatastoreService datastore;
+  private PlaceGuideRepository placeGuideRepository;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    placeGuideRepository = PlaceGuideRepositoryFactory.getPlaceGuideRepository(
+                                                                        RepositoryType.DATASTORE);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void savePlaceGuideTest() {
+    placeGuideRepository.savePlaceGuide(testPublicPlaceGuideA);
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query(DatastorePlaceGuideRepository.ENTITY_KIND);
+    PreparedQuery preparedResult = datastore.prepare(query);
+    Entity result = preparedResult.asSingleEntity();
+    Entity expected = getEntityFromPlaceGuide(testPublicPlaceGuideA);
+    assertTrue(result.equals(expected));
+  }
+
+  @Test
+  public void getAllPlaceGuides_placeGuideExists_returnPlaceGuide() {
+    List<PlaceGuide> testPlaceGuidesList = Arrays.asList(testPublicPlaceGuideA,
+                                                         testPublicPlaceGuideB,
+                                                         testPrivatePlaceGuideB,
+                                                         testPrivatePlaceGuideA);
+    saveTestPlaceGuidesEntities(testPlaceGuidesList);
+    List<PlaceGuide> result = placeGuideRepository.getAllPlaceGuides();
+    List<PlaceGuide> expected = Arrays.asList(testPublicPlaceGuideA, testPublicPlaceGuideB);
+    assertTrue(compare(expected, result));
+  }
+
+  @Test
+  public void getAllPlaceGuides_placeGuideDoesntExists_resultIsEmpty() {
+    List<PlaceGuide> result = placeGuideRepository.getCreatedPlaceGuides(CREATOR_A_ID);
```

Comment:
Here you test getCreatedPlaceGuides, although the function name shows getAllPlaceGuides.

Reply:
corrected.

Reply:
Corrected.